src/pages/quiz-page.tsx

```diff
@@ -0,0 +1,196 @@
+  RADIO_WRONG_COLOR,
+  QUESTION,
+  BUTTON_END_QUIZ_COLOR,
+  FOOTER_MARGIN,
+  BUTTON_WIDTH_DEFAULT,
+  FONT_SIZE_MEDIUM,
+} from '../contants';
+
+const styles = {
+  containerFlex: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    flexDirection: 'column',
+    height: '80vh',
+  },
+  containerHeader: {
+    flex: '1',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    height: '100%',
+    fontSize: FONT_SIZE_MEDIUM,
+    fontWeight: 'bold',
+  },
+  containerQuiz: {
+    flex: '3',
+    display: 'flex',
+    justifyContent: 'center',
+    flexDirection: 'column',
+  },
+  containerFooter: {
+    flex: '1',
+    display: 'flex',
+    justifyContent: 'center',
+    flexDirection: 'column',
+  },
+  containerButtons: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    height: '100%',
+  },
+  radio: {
+    textAlign: 'left',
+    width: '100%',
+  },
+  buttonWrapper: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  button: {
+    width: BUTTON_WIDTH_DEFAULT,
+  },
+};
+
+const QuizPage: React.FC = (): ReactElement => {
+  /**
+   * Normally this should be passed from props. This will be removed once we connected with other components..
+   */
+  const props = {
+    levelOfDifficulty: 1,
+  };
+
+  /**
+   * This mock data will be removed once the API endpoint is built.
+   */
+  const requestData = {
+    question: 'Which company belongs to 31st Engineer Battalion?',
+    choices: ['Smoking Aces', 'Dawgs', 'Cobras', 'Easy Company'],
+    correctChoiceIndex: 2,
+  };
+
+  const [score, setScore] = useState(0);
+  const [questionNumber, setQuestionNumber] = useState(1);
+  const [indexOfSelectedRadio, setIndexOfSelectedRadio] = useState(-1);
+  const [hasChooseAnswer, setHasChooseAnswer] = useState(false);
+  const [hasSubmitAnswer, setHasSubmitAnswer] = useState(false);
+
+  const isCorrectAnswer = (indexOfRadio: number) => indexOfRadio === requestData.correctChoiceIndex;
+
+  const getRadioResultColor = (index: number) => (isCorrectAnswer(index) ? RADIO_CORRECT_COLOR : RADIO_WRONG_COLOR);
+
+  const goToNextQuestion = () => {
+    setIndexOfSelectedRadio(-1);
+    setHasChooseAnswer(false);
+    setHasSubmitAnswer(false);
+    setQuestionNumber(questionNumber + 1);
+  };
+
+  const onClickRadio = (event: React.MouseEvent, { index }: ButtonProps) => {
+    if (!hasSubmitAnswer) {
+      setIndexOfSelectedRadio(index);
+      setHasChooseAnswer(true);
+    }
+  };
+
+  const onClickSkip = () => {
+    if (!hasSubmitAnswer) {
+      setScore(score - 1);
+      toast.warn(TOAST_SKIP_QUESTION, toastConfig);
+      goToNextQuestion();
+    }
+  };
+
+  const onClickSubmit = () => {
+    if (hasSubmitAnswer) {
+      goToNextQuestion();
+    } else if (hasChooseAnswer) {
+      setScore(score + (isCorrectAnswer(indexOfSelectedRadio) ? 1 : -1));
+      setHasSubmitAnswer(true);
+    }
+  };
+
+  const { levelOfDifficulty } = props;
+
```

Comment:
Do you want to separate your view and controller?

Reply:
Actually `levelOfDifficulty` should be fetched from the website URL path using `react-router` in the future. We won't pass this information as a props.

Example:
`https://us-army-quiz.netlify.app/quiz/1` -> level 1 Quiz
`https://us-army-quiz.netlify.app/quiz/2` -> level 2 Quiz

This will be updated once we setup `react-router`.